### DISCUSSION TITLE
How do I use a custom formatter file with Spotless?

### DISCUSSION THREAD

**Discussion by one-harsh** [diffplug/spotless#2034] [Q&A]
I have a custom formatter file in XML which I want to configure Spotless with in Gradle. How can I accomplish that?

**Comment by nedtwigg** [comment]
I assume this is an XML exported from Eclipse? You probably want something along these lines:

- https://github.com/diffplug/spotless/tree/main/plugin-gradle#eclipse-jdt
- https://github.com/diffplug/spotless/blob/main/ECLIPSE_SCREENSHOTS.md

**Reply by one-harsh** [reply]
I tried applying the following snippet to my gradle file, but I don't see the results I expect.
```kotlin
spotless {
    java {
        eclipse().configFile("/path/to/style.xml")
    }
}
```

A sample content for the `style.xml` file (which already existed for my project):
```xml
<code_scheme name="SomeScheme">
  <option name="RIGHT_MARGIN" value="100" />
  <JavaCodeStyleSettings>
    <option name="CLASS_COUNT_TO_USE_IMPORT_ON_DEMAND" value="99" />
    <option name="NAMES_COUNT_TO_USE_IMPORT_ON_DEMAND" value="99" />
  </JavaCodeStyleSettings>
  <XML>
    <option name="XML_LEGACY_SETTINGS_IMPORTED" value="true" />
  </XML>
</code_scheme>
```

The `./gradlew spotlessCheck` command throws the following error -
>The XML root node 'code_scheme' is not part of the supported root nodes [[properties, profiles]]

Going by the eclipse jdt doc that you linked above, my guess is that the format expected by spotless is not what I am providing. If that's correct, is there a parable API which can accomplish this task (whether directly or via some compatibility conversion).


**PS:** I am relatively new to IntelliJ/Java world.

**Reply by nedtwigg** [reply]
So this style.xml is exported from IntelliJ? You'll have to wait for #2020 to get merged and released, a week or two hopefully.
